pkg/arvo/app/invite-store.hoon

```diff
@@ -0,0 +1,186 @@
+/+  *invite-json
+|%
++$  move  [bone card]
+::
++$  card
+  $%  [%diff diff]
+      [%quit ~]
+  ==
+::
++$  state
+  $%  [%0 state-zero]
+  ==
+::
++$  state-zero
+  $:  =invites
+  ==
+::
++$  diff
+  $%  [%invite-initial invites]
+      [%invite-update invite-update]
+  ==
+--
+::
+|_  [bol=bowl:gall state]
+::
+++  this  .
+::
+++  prep
+  |=  old=(unit state)
+  ^-  (quip move _this)
+  ?~  old
+    [~ this]
+  [~ this(+<+ u.old)]
+::
+++  peek-x-all
+  |=  pax=path
+  ^-  (unit (unit [%noun (map path invitatory)]))
+  [~ ~ %noun invites]
+::
+++  peek-x-invitatory
+  |=  pax=path
+  ^-  (unit (unit [%noun (unit invitatory)]))
+  ?~  pax
+    ~
+  =/  invitatory=(unit invitatory)  (~(get by invites) pax)
+  [~ ~ %noun invitatory]
+::
+++  peek-x-invite
+  |=  pax=path
+  ^-  (unit (unit [%noun (unit invite)]))
+  ?~  pax
+    ~
+  =/  invitatory=(unit invitatory)  (~(get by invites) t.pax)
+  ?~  invitatory
+    ~
+  =/  uid=serial  (slav %uv i.pax)
+  =/  invite=(unit invite)  (~(get by u.invitatory) uid)
+  [~ ~ %noun invite]
+::
+++  peer-all
+  |=  pax=path
+  ^-  (quip move _this)
+  ?>  (team:title our.bol src.bol)
+  ::  send all updates from now on
+  :_  this
+  [ost.bol %diff %invite-initial invites]~
+::
+++  peer-updates
+  |=  pax=path
+  ^-  (quip move _this)
+  ?>  (team:title our.bol src.bol)
+  ::  send all updates from now on
+  [~ this]
+::
+++  peer-invitatory
+  |=  pax=path
+  ^-  (quip move _this)
+  ?>  (team:title our.bol src.bol)
+  =/  inv=(unit invitatory)  (~(get by invites) pax)
+  ?~  inv  !!
+  :_  this
+  [ost.bol %diff %invite-update [%invitatory u.inv]]~
+::
+++  poke-json
+  |=  =json
+  ^-  (quip move _this)
+  ?>  (team:title our.bol src.bol)
+  (poke-invite-action (json-to-action json))
+::
+++  poke-invite-action
+  |=  action=invite-action
+  ^-  (quip move _this)
+  ?>  (team:title our.bol src.bol)
+  ?-  -.action
+      %create
+    (handle-create action)
+  ::
+      %delete
+    (handle-delete action)
+  ::
+      %invite
+    (handle-invite action)
+  ::
+      %accept
+    (handle-accept action)
+  ::
+      %decline
+    (handle-decline action)
+  ::
+  ==
+::
+++  handle-create
```

Comment:
Why not auto-create these the first time an `%invite` happens on some path? I'm not sure what use an explicit action & diff for "we have a new invites collection, but there's nothing in it" serves.

Reply:
We don't want to auto-create because we don't want to have to clean up random invite paths. We want individual `invitatories` to be managed by specific apps which display interfaces / have affordances for cleaning out invites.

Reply:
It would be difficult to manage some disparate number of paths that could have invites being sent to them.

Reply:
> We want individual `invitatories` to be managed by specific apps

For `invitatory`, is the distinction between "does not exist" and "does not contain any invites" meaningful?

I'm arguing that it isn't. Readers of the invite-store _shouldn't_ care about that distinction, and since _every single invite-sending application_ will just be doing "create this" before doing "add things to it", the store should just generalize that into itself.

If you're saying the distinction _is_ meaningful, then yes, manual creation is justified. If so, let me know your case for this.

(Related, adding to `invitatory` path that doesn't exist currently just silently no-ops. Seems a bit dirty. Noteworthy that auto-creating as discussed above would get rid of the no-op.)

Reply:
The distinction is meaningful, because if you send to an `invitatory` that doesn't exist, you should not store the invite.

Reply:
That's true, but tautological. The question is: should it be meaningful? The answer depends on a) what invites are (for), and b) how clients/consumers are expected to interact with the invitiverse.